verify/cert.go

```diff
@@ -103,10 +125,35 @@ func (cv *CertVerifier) VerifyMerkleProof(inclusionProof []byte, root []byte,
 }
 
 // fetches a block number provided a subtraction of a user defined conf depth from latest block
-func (cv *CertVerifier) getConfDeepBlockNumber() (*big.Int, error) {
-	blockNumber, err := cv.ethClient.BlockNumber(context.Background())
+func (cv *CertVerifier) getConfDeepBlockNumber(ctx context.Context) (*big.Int, error) {
+	if cv.waitForFinalization {
+		// https://github.com/ethereum/execution-apis/blob/4140e528360fea53c34a766d86a000c6c039100e/src/eth/block.yaml#L61
+		var header = types.Header{}
+		err := cv.ethClient.Client().CallContext(ctx, &header, "eth_getBlockByNumber", "finalized", false)
```

Comment:
assuming this is encoded to the raw params like [`finalized, `false`] where the boolean indicates that we aren't collecting txs for the block? We may wanna provide more explicit detailing of this value for easier readability long-term

Reply:
[7069e13](https://github.com/Layr-Labs/eigenda-proxy/pull/192/commits/7069e13e252623e6e9b163e7e8b628e7c662e2bb)